src/MouseInfoProvider/index.js

```diff
@@ -1,125 +1,82 @@
-import React, { Component } from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import MouseInfoContext from './context';
 
-class MouseInfoProvider extends Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      animationScheduled: false,
-      x: 0,
-      y: 0,
-      xDifference: 0,
-      yDifference: 0,
-      xDirection: '',
-      yDirection: '',
-      xPercentage: 0,
-      yPercentage: 0,
-      totalPercentage: 0,
-      isInViewport: false,
-      eventsFired: 0,
-    };
-  }
-
-  componentDidMount() {
-    document.addEventListener('mousemove', this.requestAnimation);
-    document.addEventListener('mouseenter', () => this.setViewportStatus(true));
-    document.addEventListener('mouseleave', () => this.setViewportStatus(false));
-  }
-
-  componentWillUnmount() {
-    document.removeEventListener('mousemove', this.requestAnimation);
-    document.removeEventListener('mouseenter', this.setViewportStatus);
-    document.removeEventListener('mouseleave', this.setViewportStatus);
-  }
-
-  setViewportStatus = (status) => {
-    this.setState({ isInViewport: status });
-  }
-
-  updateMouseInfo = (e) => {
-    const {
-      x: prevMouseX,
-      y: prevMouseY,
-      xDirection: prevXDirection,
-      yDirection: prevYDirection,
-    } = this.state;
-
-    const currentMouseX = e.clientX;
-    const currentMouseY = e.clientY;
-
-    const xDifference = currentMouseX - prevMouseX;
-    const yDifference = currentMouseY - prevMouseY;
-
-    const xPercentage = Number((currentMouseX / window.innerWidth).toFixed(3));
-    const yPercentage = Number((currentMouseY / window.innerHeight).toFixed(3));
-    const totalPercentage = Number(((xPercentage + yPercentage) / 2).toFixed(3));
-
-    const xDirection = xDifference > 0 ? 'right' : xDifference < 0 ? 'left' : prevXDirection;
-    const yDirection = yDifference > 0 ? 'down' : yDifference < 0 ? 'up' : prevYDirection;
-
-    this.setState({
-      animationScheduled: false,
-      x: currentMouseX,
-      y: currentMouseY,
-      xDifference,
-      yDifference,
-      xDirection,
-      yDirection,
-      xPercentage,
-      yPercentage,
-      totalPercentage,
-    });
-  };
+const MouseInfoProvider = (props) => {
+  const { children } = props;
+
+  const [isAnimationScheduled, setIsAnimationScheduled] = useState(false);
```

Comment:
[Refer to this review in `react-node-position`](https://github.com/trouble/react-node-position/pull/2#discussion_r359126029).

Reply:
I have adjusted the code to group states that get updated at the same time, into a single state object now.